Add Reviewers to pr view in CLI

doi-t opened the pull request:
This PR is a part of #663 (`pr view`).

~(This is a nested PR. #748 needs to be merged at first.)~ Merged.

[The query updates](https://github.com/doi-t/cli/compare/85d8f80...doi-t:8fca78f) for `Reviewers` is in #748.

~The design must be reviewed~ The design has been reviewed https://github.com/cli/cli/issues/663#issuecomment-610644914. The author of a PR won't be included in the `Reviewers` list as GitHub.com does not include it ([this](https://github.com/microsoft/vscode-pull-request-github/blob/7fbd5ec/src/github/pullRequestOverview.ts#L98-L153) is the reference implementation).

When there is a requested reviewer:
![Screen Shot 2020-04-08 at 0 00 25](https://user-images.githubusercontent.com/5877477/78684940-0c4b3a00-792c-11ea-8c14-78fb48ec8180.png)

When there are multiple reviews:
![Screen Shot 2020-04-07 at 23 59 59](https://user-images.githubusercontent.com/5877477/78684931-0a817680-792c-11ea-9ce8-9018bf32c31c.png)

When there is no reviews and requested reviewer (No metadata too):
![Screen Shot 2020-04-08 at 0 00 40](https://user-images.githubusercontent.com/5877477/78684938-0bb2a380-792c-11ea-928a-625d819c75b2.png)

## Todos
- [x] Implement a parser for `Reviews` and `ReviewRequests`
- [x] sort reviewer list
- [x] Write unit tests

doi-t commented:
@ampinsk 
I've added colors to the reviewer's state following https://github.com/cli/cli/issues/663#issuecomment-610644914.

This is a random pick example that includes multiple reviewer's states. I put completed reviews before review requests and sort the reviewer's name alphabetically. As far as I can observe, GitHub.com does further sorting (top > `Changes requested` > `Commented` > `Approved` > `Requested` > bottom & alphabetical sort of reviewer's name???) but I guess this can be a starting point of `gh`. 🤔 
<img width="638" alt="Screen Shot 2020-04-10 at 1 08 33" src="https://user-images.githubusercontent.com/5877477/78916139-fc685d00-7ac7-11ea-9017-51fab2a68897.png">

This is an example of `Commened` state.
<img width="588" alt="Screen Shot 2020-04-10 at 1 10 33" src="https://user-images.githubusercontent.com/5877477/78916218-1bff8580-7ac8-11ea-8a44-ea54c336ee45.png">

@mislav @vilmibm 
This PR is also ready for code review once https://github.com/cli/cli/pull/748 gets merged!

ampinsk commented:
@doi-t yes this is totally great for now! I definitely don't think we need to over optimize right now, this is exactly what we need to start :)

doi-t commented:
Solved the conflict and made additional refactoring. Now it's ready for code review.

tierninho commented:
Tested this change and it functions well. 👍 

I like that that reviewers and the type of review are both in alpha order, however I noticed that there are actually four ways we can organize this info. I am not sure of the importance of this, but wanted to point it out in case we missed it.

- (Current version) Alpha order by username and then by review-type, like so:
![Screen Shot 2020-04-13 at 11 38 19 AM](https://user-images.githubusercontent.com/14828183/79163708-4c227f00-7d7b-11ea-8973-567d49370a12.png)
- Chronologically beginning with the first request for a review. Tom was requested first, Jerry was second. See sidebar in pic below.
- Chronologically beginning with an actual review. Jerry beat Tom to reviewing the PR. See timeline in pic below.
- We set an alternative particular order, tbd.

![Screen Shot 2020-04-13 at 11 29 41 AM](https://user-images.githubusercontent.com/14828183/79163207-51cb9500-7d7a-11ea-88b1-cc7ceec5b0ac.png)

doi-t commented:
Thank you for the double-check!

> - Chronologically beginning with the first request for a review. E.g., Tom was requested first, Jerry was second. See sidebar in pic below.
> - Chronologically beginning with an actual review. E.g., Jerry beat Tom to reviewing the PR. See timeline in pic below.

Since an actual review can happen without a review request and "reviews" and "review requests" are managed differently by GitHub, I guess we need to combine them if we go with this idea (e.g. `the first review (Approved), the second review (Changes requested), the first review request (Requested), the second review request (Requested)`). I'm not sure if we can get the timestamp for "review requests". 🤔

ampinsk commented:
I'm not too worried about the ordering for this first version! I think whatever's simplest is great, and we can always iterate on this detail in the future if we find people are getting confused by this or are needing more nuanced ordering 😄

mislav commented on test/fixtures/prViewPreviewWithMetadataByNumber.json:
```diff
@@ -10,6 +10,39 @@
+            "nodes": [
+                {
+                    "requestedReviewer": {
+                        "__typename": "user",
```
I think it might be valuable to also display teams among requested reviewers. But I also think that this is something that could just as well be a follow-up.

doi-t replied:
Let me create another follow-up PR for `team`! Since the key is different from `user`, I guess I need to tweak the sorting logic along with a query update.

doi-t replied:
@mislav 
I've added the team reviewer support https://github.com/cli/cli/pull/762/commits/d70358ea3445cec9d2f341c0448abab966610d8b to this PR.

The following test results are made by below permissions (`read:org` works 👍 ).
```diff
-       q.Set("scope", "repo")
+       q.Set("scope", "repo read:org")
```

It seems that the requested team reviewer will be gone once one of the team members takes an action. The requested team will not be shown in the review results too because a result (`Comment`/`Approved`/`Request changes`) is always recorded with user account as far as I can see.

![Screen Shot 2020-04-22 at 22 05 45](https://user-images.githubusercontent.com/5877477/79985362-7562ae00-84e5-11ea-92a0-dc62bff2974c.png)


I also realized that `ghost` account does not have `Login` value. I added a workaround to show `ghost`'s review results https://github.com/cli/cli/pull/762/commits/67907c8b004fc5331900119b9882ca481b6cc68d.

![Screen Shot 2020-04-22 at 22 17 05](https://user-images.githubusercontent.com/5877477/79986457-07b78180-84e7-11ea-98ac-d19a00c34b6f.png)

I think this edge case is happening for other commands which display a user account. 🤔

doi-t commented:
I've merged `master` into this branch to include #786 and confirmed the work.

![Screen Shot 2020-04-30 at 0 37 19](https://user-images.githubusercontent.com/5877477/80615577-d1e04300-8a7a-11ea-87b7-9ede1d29643a.png)

This PR is now ready for re-review and merge. https://github.com/cli/cli/pull/762#discussion_r412978672 is additional changes since the last code review.

By the way, `gh` asked me to do auth again as I expected. However, it didn't work so that I needed to remove `~/.config/gh/config.yml`.

```
$ gh pr view 1 --repo cli-test-org/cli-test
Notice: additional authorization required
Press Enter to open github.com in your browser...
Authentication complete. Press Enter to continue...

graphql error: 'Your token has not been granted the required scopes to execute this query. The 'name' field requires one of the following scopes: ['read:org', 'read:discussion'], but your token has only been granted the: ['repo'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens.'
```
